Package.swift

```diff
@@ -30,30 +33,106 @@ let package = Package(
-
+        // Targets are the basic building blocks of a package. A target can define a module or a test suite.
+        // Targets can depend on other targets in this package, and on products in packages this package depends on.
         .target(
-            name: "MaaS",
-            dependencies: ["MaasCore", "Swappable", "MaasTheme"],
-            path: "ios/Sources/MaaS"),
-
+            name: "Maas",
+            dependencies: ["Swappable"],
+            path: "ios/Sources/Maas"),
+        .target(
+            name: "Examples",
+            dependencies: [
+                "MaasNearbyTransit",
+                "MaasRouteSearch"
+            ],
+            path: "ios/Sources/Examples"
+        ),
         .testTarget(
             name: "MaasTests",
-            dependencies: ["MaaS", "SnapshotTesting", "Quick"],
+            dependencies: [
+                "Maas",
+                "MaasNearbyTransit",
+                "MaasRouteSearch",
+                "SnapshotTesting",
+                "Quick"
+            ],
             path: "ios/Tests/MaasTests",
-            exclude: ["__Snapshots__"]),
-
-        .target(
-            name: "MaasTheme",
-            dependencies: ["MaasCore"],
-            path: "ios/Sources/MaasTheme"),
-    ]
+            exclude: ["__Snapshots__"]), 
+    ] + environment.targets 
 )
+
+// MARK: - Setup Extension
+
+private extension Environment {
+
+    var products: [Product] {
+        switch self {
+        case .development:
+            return [
+                // --
+            ]
+        case .production:
+            return [
+                .library(
+                    name: "MaasCore",
+                    targets: ["MaasCore"]),
+               .library(
+                   name: "MaasNearbyTransit",
+                   targets: ["MaasNearbyTransit"]),
+               .library(
+                   name: "MaasRouteSearch",
+                   targets: ["MaasRouteSearch"]),
+           ]
+        }
+    }
+
+    var targets: [Target] {
+        switch self {
+        case .development:
+            return [
+                // --
+            ]
+        case .production:
+            return [
+                .binaryTarget(
+                    name: "MaasCore",
+                    path: "ios/MaasCore/Sources/MaasCore/Core.xcframework"),
+                .target(
+                    name: "MaasTheme",
+                    dependencies: ["Swappable"],
+                    path: "ios/MaasCore/Sources/MaasTheme"),
+                .target(
+                    name: "MaasComponents",
+                    dependencies: ["MaasTheme", "Swappable"],
+                    path: "ios/MaasCore/Sources/MaasComponents"), 
+                .target( 
+                    name: "MaasNearbyTransit",
+                    dependencies: ["MaasCore", "MaasTheme", "MaasComponents", "Swappable"],
+                    path: "ios/MaasNearbyTransit/Sources",
+                    exclude: ["Package.resolved"]),
```

Comment:
Should `Package.resolved` be excluded from the other targets (`MaasTheme`, `MaasComponents`) too? Or do they not produce such a file?

Reply:
`Package.resolved` appears when we're using a non local package as dependency. So at this moment `MaasNearbyTransit` has it's own `Swappable` dependency. But when we're using `MaasNearbyTransit` as a source not a package we're getting warning that tells us to exclude `Package.resolved`.